### ISSUE TITLE
Switching between mobile and desktop layout

### ISSUE THREAD

**Issue by pearcemichal** [grubersjoe/slide-menu#8]
My site has a different layout for desktop and mobile, and I am only using slide-menu for the mobile layout. I am currently using a page width check on load to either initialize slide-menu or not, however it would be nice if I could dynamically enable and disable slide-menu on an element. 

dynamically enabling the menu's isn't an issue, as I can just call the code to create the menus on page resize, the issue is disabling them. as far as I can tell, there is no native ability to undo the DOM changes that new SlideMenu() creates. 

is this a function that could be added?

**Comment by grubersjoe** [comment]
Sorry for the late reply. I've looked into this and added the feature in the [feature/destroy branch](https://github.com/grubersjoe/slide-menu/tree/feature/destroy).

Could you test it, if this works as expected? 

This adds a new `menu.destroy()` method, which removes all added DOM elements. The inline styles of the menu itself will not be touched, however. I'm not sure, whether these should also be removed. What do you think?

**Comment by grubersjoe** [comment]
Ah, I see. I've made the mistake to set the `innerText` attribute instead of `innerHtml` for the link "decorations". I've fixed it and iIt should work as before now.

Your code is very close to what I did:

* remove `submenuLinkAfter` and `submenuLinkBefore`
* remove all backlinks
* remove the wrapper DOM element
* remove inline styles
* delete the references to the SlideMenu instance saved as property of the DOM element

However, I'm not deleting the class name `slide-menu`, since this would alter the HTML (it was already there before SlideMenu was added). If you need this, you can simply to it yourself after calling `destroy()`.

**Comment by vcoppolecchia** [comment]
Hi and thank you for this method (simply awesome). 
Do you have any idea of when you'll merge this in `master` and publish it on npm?

**Comment by grubersjoe** [comment]
It's merged and published to npm now.

What kind of errors do you get? I've added a part to the demo page to test the events. The `sm.open-after` event was broken indeed, but I fixed it in the recent version. Now, everything works as expected (at least in the demo).

Have you added the event listener to the correct HTML element (the menu element)?

**Comment by vcoppolecchia** [comment]
@grubersjoe thank you for publishing the new version on npm, just updated and `.destroy()` method seems to work great! 😊

> What kind of errors do you get?

Regarding custom events: before the update I was on version `1.0.13` and your plugin's events were working fine by doing this:
```javascript
window.mobileMenu = new SlideMenu(menuContainer, { /* my options here... */ });
window.mobileMenu.addEventListener('sm.open', function(e) { /* Do things here... */ });
```
Now I'm on version `1.1.3` and that kind of `addEventListener` started throwing error ("_Uncaught TypeError: window.mobileMenu.addEventListener is not a function_"). Based on your suggestions I changed the `addEventListener` to:
```javascript
menuContainer.addEventListener('sm.open', function(e) { /* Do things here... */ });
```
and now events are working fine again for me.

### Other things I've noticed (and that are not clear to me)
![image](https://user-images.githubusercontent.com/900011/52569534-b6573100-2e11-11e9-8baf-34e542f90bf1.png)
- **I noticed** that when I click the little arrow to navigate back (`backLinkBefore` option), the menu doesn't navigate back; in the opposite direction (forward arrow inside `<span class="slide-menu__decorator">...</span>`) this is working as expected.
- **Is not clear to me** how to use `keycodeClose` and `keycodeOpen` options, can you please provide an example in the [demo page](https://grubersjoe.github.io/slide-menu/)?

Should I open new issues about these things? What do you think?

Thank you again!

**Comment by grubersjoe** [comment]
Glad you got it working again :+1:. I'm going to update the README to make it clearer, how to attach event listeners to a specific menu. This is documented badly right now.

> that when I click the little arrow to navigate back (`backLinkBefore` option), the menu doesn't navigate back

You're right, this is not how it should be. I'm going to fix this as soon as possible.

The key codes, are the ones described on this [MDN page](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values). Example:

```js
new SlideMenu(menuRightElem, {
    keyClose: 'Escape',
    keyOpen: 'o',
});


```
This page might also be helpful: https://keycode.info (press any key, it's the value at the most left)

**Comment by vcoppolecchia** [comment]
> You're right, this is not how it should be. I'm going to fix this as soon as possible.

For everybody here I managed to solve the problem of the backlink pretty easily by setting the CSS property `pointer-events: none` to the DOM element containing the "back" arrow. As often times happens, [this StackOverflow question](https://stackoverflow.com/questions/26402033/) has proved its utility! 😄

Tested on the latest Firefox (65) and Chrome (72).
